### ISSUE TITLE
System.Drawing.Imaging.ImageAttributes.SetOutputChannelColorProfile() Throws System.Runtime.InteropServices.ExternalException on netcoreapp - does not on netfx (#20711)

### ISSUE THREAD

**Issue by KostaVlev** [dotnet/corefx#22367]
The test:

``` c#
[ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
public void SetOutputChannelColorProfile_Null_ThrowsArgumentNullException()
{
     using (var imageAttr = new ImageAttributes())
    {
        Assert.Throws<ArgumentNullException>(() => imageAttr.SetOutputChannelColorProfile(null));
        Assert.Throws<ArgumentNullException>(() => imageAttr.SetOutputChannelColorProfile(null, ColorAdjustType.Default));
    }
}
```

**Expected:** `ArgumentNullException`

**Actual:** 
`netfx` - `ArgumentNullException`
`netcoreapp` - `System.Runtime.InteropServices.ExternalException: A generic error occurred in GDI+.`

(#20711)

**Comment by hughbe** [comment]
Another case of missing `Path.GetFullPath`. The netfx code is:
```cs
public void SetOutputChannelColorProfile(String colorProfileFilename,
                                          ColorAdjustType type)
{
    IntSecurity.DemandReadFileIO(colorProfileFilename); // MISSING.

    int status = SafeNativeMethods.Gdip.GdipSetImageAttributesOutputChannelColorProfile(
                                new HandleRef(this, nativeImageAttributes),
                                type,
                                true,
                                colorProfileFilename);
                    
    if (status != SafeNativeMethods.Gdip.Ok)
            throw SafeNativeMethods.Gdip.StatusException(status);
}
```

`IntSecurity.DemandReadFileIO(colorProfileFilename)` is not available in .NET Core, so we just deleted it.

The fix would be to replace `IntSecurity.DemandReadFileIO(colorProfileFilename)` with a simple call to `Path.GetFullPath(colorProfileFilename)`

Take a look at https://github.com/dotnet/corefx/issues/21558

The fix should be accompanied by tests for the empty string as well as a path longer than the max path value (> 260 chars)

@KostaVlev feel like fixing this once you've submitted tests?

**Comment by KostaVlev** [comment]
@hughbe  sure.... I will be ready with the tests shortly too.

**Comment by KostaVlev** [comment]
I updated `ImageAttributes.SetOutputChannelColorProfile()`:

```c#
public void SetOutputChannelColorProfile(String colorProfileFilename, ColorAdjustType type)
{
    Path.GetFullPath(colorProfileFilename);

    int status = SafeNativeMethods.Gdip.GdipSetImageAttributesOutputChannelColorProfile(
                                        new HandleRef(this, nativeImageAttributes),
                                        type,
                                        true,
                                        colorProfileFilename);

    if (status != SafeNativeMethods.Gdip.Ok)
                throw SafeNativeMethods.Gdip.StatusException(status);
}
```

As @hughbe  suggested I added this line of code `Path.GetFullPath(colorProfileFilename);` which will validate the `colorProfileFilename` and this fixes the current issue when the path is Null and we get the correct exception. 

but :)

**Test for path too long fails:**

```c#
[ConditionalFact(Helpers.GdiplusIsAvailable)]
public void SetOutputChannelColorProfile_InvalidPath_ThrowsPathTooLongException()
{
    string fileNameTooLong = new string('a', 261);
    using (var imageAttr = new ImageAttributes())
    {
        Assert.Throws<PathTooLongException>(() => 
            imageAttr.SetOutputChannelColorProfile(fileNameTooLong));
        Assert.Throws<PathTooLongException>(() => 
            imageAttr.SetOutputChannelColorProfile(fileNameTooLong, ColorAdjustType.Default));
    }
}
```
**Expected:** `System.IO.PathTooLongException`

**Actual:**
netfx `System.IO.PathTooLongException`
netcoreapp `System.OutOfMemoryException`

I can't figure out why `necoreapp` throws different exception  when `Path.GetFullPath()` should return  `System.IO.PathTooLongException` when the string length is longer then 260 charters. 

[GetFullPath() docs](https://msdn.microsoft.com/en-us/library/system.io.path.getfullpath(v=vs.110).aspx)

**Comment by mellinoe** [comment]
@KostaVlev `Path.GetFullPath` was changed in .NET Core 2.0. It no longer pre-emptively throws a `PathTooLongException`. So this test will need to be changed to account for that. Most likely, it should be removed, or made netfx-specific. I'm skeptical that netfx-specific tests have any value for us in this repository, though, so I'd just delete it.

**Comment by mellinoe** [comment]
@KostaVlev I'm going to delete the test expecting a PathTooLongException in an upcoming PR.

**Comment by KostaVlev** [comment]
Fixed by  #22941
